processor/src/main/java/org/mapstruct/ap/internal/model/beanmapping/SourceReference.java

```diff
@@ -80,19 +93,23 @@ public BuilderFromMapping typeFactory(TypeFactory typeFactory) {
             return this;
         }
 
+        public BuilderFromMapping sourceName(String sourceName) {
+            this.sourceName = sourceName;
+            return this;
+        }
+
         public SourceReference build() {
 
-            String sourceName = mapping.getSourceName();
             if ( sourceName == null ) {
                 return null;
             }
 
             String sourceNameTrimmed = sourceName.trim();
-            if ( !sourceName.equals( sourceNameTrimmed ) ) {
+            if ( !sourceName.equals( sourceNameTrimmed ) && messager != null ) {
```

Comment:
How is it possible that the `messager` is `null`? We should always pass it, otherwise we can't print messages

Reply:
When pushing the source reference I'm not interested in error messaging. It is part of processing not part of analysing the source model. I'm just interested if we have a valid result or not.

Reply:
I don't really like that. Even if you are not interested in the error messaging the conditions would always evaluate to `true`, right? Currently it is really easy to forget the messager and not log an error for that

Reply:
Duplicating all that code is also not an option

Reply:
Duplicating all that code is also not an option

Reply:
Why not just pass the messager there as well? When doing the push there should be no errors in any way, right?

Reply:
ok.. resolved.